appwrite/appwrite
# Feat: Wildcard hostname

## What does this PR do?

- We now support `*.vercel.all` or `*.*.gitpod.io` or even simply `*` hostnames
- Hostname is now properly validated when creating/updating. You now cannot enter an invalid hostname
- In UI, when platform update fails, refresh list so we don't have incorrect data in list

## Test Plan


1. Don't allow port and path:


https://user-images.githubusercontent.com/19310830/164199120-6b01218d-36ba-4a74-b2d3-6fc90aeaa5a4.mp4

2. Ensure wildcard can be inserted properly:

![CleanShot 2022-04-20 at 11 37 39](https://user-images.githubusercontent.com/19310830/164199177-aeadaf0c-397c-4f01-8bc8-9247676ce6e1.png)

3. Ensure wildcards are actually supported:

![CleanShot 2022-04-20 at 11 43 25](https://user-images.githubusercontent.com/19310830/164200318-0f54d0b4-1373-453a-82cb-78c3f6ef0495.png)

(I was able to create and read an anonymous session from `https://8080-meldiron-almostredditpla-1cocjbj09gp.ws-eu34xl.gitpod.io/index2.html` with the hostname was `*.*.gitpod.io` or `*` or `*.*.gitpod.*`)

## Related PRs and Issues

- Outdated closed PR: https://github.com/appwrite/appwrite/pull/2721
- Issue: https://github.com/appwrite/appwrite/issues/2939

### Have you read the [Contributing Guidelines on issues](https://github.com/appwrite/appwrite/blob/master/CONTRIBUTING.md)?

Yes
